port/esp32_port.c

```diff
@@ -142,18 +142,18 @@ esp_loader_error_t loader_port_read(uint8_t *data, uint16_t size, uint32_t timeo
 // assert reset pin for 50 milliseconds.
 void loader_port_enter_bootloader(void)
 {
-    gpio_set_level(s_gpio0_trigger_pin, 0);
+    gpio_set_level(s_gpio0_trigger_pin, SERIAL_FLASHER_BOOT_INVERT ? 1 : 0);
```

Comment:
```suggestion
    gpio_set_level(s_gpio0_trigger_pin, CONFIG_SERIAL_FLASHER_BOOT_ACTIVE_LEVEL);
```

Reply:
esp32_port.c refers to `SERIAL_FLASHER_RESET_HOLD_TIME_MS` as such without `CONFIG_` prefix.
Is there something I don't understand why this new option would behave differently?

This same question goes for the other comments too.

Reply:
Hi @shiona, this is due to the fact that KConfig options do not get simply passed as defines to the build in ESP-IDF. KConfig options are processed and then CMake variables are created from them, which have to be handled in the build system.

These CMake variables will have the `CONFIG` prefix, unlike CMake variables for non-esp ports.

You can see that we have created a [convenience macro](https://github.com/espressif/esp-serial-flasher/blob/master/CMakeLists.txt#L14) for adding options which handles those differences and also the `n` and `y` booleans from KConfig, please use this to add the option.